app/scripts/views/mixins/signin-mixin.js

```diff
@@ -55,19 +56,35 @@ define(function (require, exports, module) {
     },
 
     onSignInSuccess: function (account) {
-      if (! account.get('verified')) {
+      if (account.get('verified') && account.get('challenge') === Constants.SIGNIN_EMAIL) {
```

Comment:
I wonder if we need to have `verified` here? Seems like its original function was to say if an email has been verified not really if a token has been. I think using `tokenVerified` with `challengeReason` and `challengeMethod` might be a better choice. `tokenVerified` is sent in login response.